docs/contributing/semgrep-contributing.md

```diff
@@ -127,6 +127,38 @@ make rebuild
 
 See the Makefile in `cli/`
 
+## Adding python packages to `semgrep`
+
+While developing you may find that you would like to add a package dependency. Semgrep uses mypy to do static type-checking of its python code, so when you add a new python package, you will also need to add typing stubs for that package. You can add the package you want along with its typing package in three steps. For example, suppose you would like to add the package `pyyaml` to Semgrep.
+
+First, you will need to install the corresponding package with typing stubs. For our `pyyaml` example, the corresponding package is `types-pyyaml`.
+
+```
+pipenv install --dev types-pyyaml
+```
+
+Here we use --dev to specify that this package is needed for development but not in production. This command will update `cli/Pipfile` with the typing stubs package, and will add both the typing stubs and the package itself to your `Pipfile.lock`. This should allow you to import the package in your code. For example, `import yaml as pyyaml`.
+
+Next, you will need to manually add the typing stubs package to `.pre-commit-config.yaml` so that the pre-commit mypy hook can find the package.
+
+```
+      - id: mypy
+        additional_dependencies: &mypy-deps
+          - ...
+          - types-PyYAML
+```
+
+Finally, you will need to manually add the original package to `cli/setup.py` in the `install_requires` list variable. You can find the version number either in the `Pipfile.lock` changes or by looking up online the most recent major version of the package.
+
+```
+install_requires = [
+   ...
+   "pyyaml~=6.0",
+]
+```
+
+This change allows docker to find the package when building. Without this change, if you create a pull request you will see a CI job called `build docker image` fail with a `ModuleNotFoundError`, indicating that it is unable to find your package.
```

Comment:
This isn't just to allow docker to find the package while building, this is so the package is a dependency of published semgrep

Reply:
ok, fixed